contracts/collect/AaveFeeCollectModule.sol

```diff
@@ -0,0 +1,258 @@
+     *  2. Ensuring the current timestamp is less than or equal to the collect end timestamp
+     *  2. Ensuring the collect does not pass the collect limit
+     *  3. Charging a fee
+     */
+    function processCollect(
+        uint256 referrerProfileId,
+        address collector,
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external override onlyHub {
+        if (_dataByPublicationByProfile[profileId][pubId].followerOnly)
+            _checkFollowValidity(profileId, collector);
+
+        uint256 endTimestamp = _dataByPublicationByProfile[profileId][pubId].endTimestamp;
+        uint256 collectLimit = _dataByPublicationByProfile[profileId][pubId].collectLimit;
+        uint256 currentCollects = _dataByPublicationByProfile[profileId][pubId].currentCollects;
+
+        if (currentCollects >= collectLimit && collectLimit != 0) {
+            revert Errors.MintLimitExceeded();
+        } else if (block.timestamp > endTimestamp && endTimestamp != 0) {
+            revert Errors.CollectExpired();
+        } else {
+            _dataByPublicationByProfile[profileId][pubId].currentCollects = ++currentCollects;
+            if (referrerProfileId == profileId) {
+                _processCollect(collector, profileId, pubId, data);
+            } else {
+                _processCollectWithReferral(referrerProfileId, collector, profileId, pubId, data);
+            }
+        }
+    }
+
+    /**
+     * @notice Returns the publication data for a given publication, or an empty struct if that publication was not
+     * initialized with this module.
+     *
+     * @param profileId The token ID of the profile mapped to the publication to query.
+     * @param pubId The publication ID of the publication to query.
+     *
+     * @return The ProfilePublicationData struct mapped to that publication.
+     */
+    function getPublicationData(uint256 profileId, uint256 pubId)
+        external
+        view
+        returns (ProfilePublicationData memory)
+    {
+        return _dataByPublicationByProfile[profileId][pubId];
+    }
+
+    function _processCollect(
+        address collector,
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) internal {
+        uint256 amount = _dataByPublicationByProfile[profileId][pubId].amount;
+        address currency = _dataByPublicationByProfile[profileId][pubId].currency;
+        _validateDataIsExpected(data, currency, amount);
+
+        (address treasury, uint16 treasuryFee) = _treasuryData();
+        address recipient = _dataByPublicationByProfile[profileId][pubId].recipient;
+        uint256 treasuryAmount = (amount * treasuryFee) / BPS_MAX;
+        uint256 adjustedAmount = amount - treasuryAmount;
+
+        _transferFromAndDepositToAaveIfApplicable(currency, collector, recipient, adjustedAmount);
+        IERC20(currency).safeTransferFrom(collector, treasury, treasuryAmount);
```

Comment:
It worth to add this here to avoid the overhead when no treasury fee set (like it's set right now)
```suggestion
        if (treasuryAmount > 0) {
            IERC20(currency).safeTransferFrom(collector, treasury, treasuryAmount);
        }
```